crates/data_components/src/graphql/client.rs

```diff
@@ -327,23 +375,28 @@ impl PaginationParameters {
                         let json_pointer = data_field
                             .map(|f| format!("/data{current_path}/{:?}", f.name).replace('"', ""));
 
-                        match parent_field.try_into() {
-                            Ok(pag_arg) => {
-                                return (
-                                    Some(PaginationParameters {
-                                        resource_name: format!("{:?}", parent_field.name)
-                                            .replace('"', ""),
-                                        args: pag_arg,
-                                        page_info_path: Some(new_path),
-                                    }),
-                                    json_pointer,
-                                );
-                            }
+                        let pag_args = match TryInto::<PaginationArgument>::try_into(parent_field) {
+                            Ok(pag_args) => pag_args,
                             Err(e) => {
-                                tracing::warn!("Invalid pagination argument: {e}");
+                                tracing::warn!("Invalid pagination argument from field: {e}");
                                 return (None, None);
                             }
+                        };
+
+                        // Check [`PaginationArgument`] and `pageInfo` fields are consistent.
+                        if let Err(e) = pag_args.validate_page_info(field) {
+                            tracing::warn!("GraphQL query has pagination specified ({pag_args}), but invalid pagination fields: {e}");
+                            return (None, None);
                         }
+
+                        return (
+                            Some(PaginationParameters {
+                                resource_name: format!("{:?}", parent_field.name).replace('"', ""),
```

Comment:
Why is this replace needed?

Reply:
The `{:?}` outputs `'"field_name"'`, not `'field_name'`. There's no way I can find to render the `Text` easily otherwise.

Reply:
I pushed up a commit that fixes it